src/app/documentation/_components/DocumentationContent.tsx

```diff
@@ -24,7 +29,37 @@ export default function DocumentationContent({
   const { i18n } = useClientTranslation()
   const path = decodeURI(slugs.join('/'))
 
-  const { engine } = useEngine()
+  const { user, getCurrentSimulation } = useUser()
+
+  const lang = useLocale()
+
+  const { data: rules } = useRules({
+    lang: lang || 'fr',
+    region: supportedRegions[user.region?.code] ? user.region.code : 'FR',
+    isOptim: false,
+  })
+
+  const currentSimulation = getCurrentSimulation()
+  const situation = currentSimulation?.situation
+
+  const engine = useMemo<Engine | null>(
+    () => (rules ? new Engine(rules as Rules) : null),
+    [rules]
+  )
+
+  //TODO: this is shit
+  useEffect(() => {
+    if (engine && situation) {
+      const rules = Object.keys(engine.getParsedRules())
+
+      const safeSituation: Situation = safeGetSituation({
+        situation,
+        everyRules: rules,
+      })
+
+      engine.setSituation(safeSituation as any)
+    }
+  }, [engine, situation])
```

Comment:
Pourquoi "code degueu" ?
De ce que je comprends, l'idée ici (à juste titre) est :
- D'avoir les règles non optimisées dans la doc
- Que la doc corresponde à la situation de la simulation courante

Je vois le commentaire initial "This is shit", c'était de toi ou @florianpanchout ? C'était pour quelle raison ? Est ce qu'on a une piste pour améliorer ce bout de code ?

Reply:
Ahah le commit est de moi et le commentaire de Florian, pas un pour rattraper l'autre. Non c'est juste que ça fait beaucoup de code pour ce qu'on cherche à faire ; à voir si c'est améliorable, j'ai l'intuition que oui mais je n'y ai pas réfléchi. Dans l'idéal aussi on pourrait extraire ce code dans un hook pour rendre ça plus propre aussi ! N'y voit aucune attaque personnelle en tout cas, je me souviens pas qui a codé ça mais il me semblait que c'était @florianpanchout

Reply:
C'est juste que de mon point de vue c'est pas si horrible, c'est un peu long effectivement mais au moins on a le bon comportement ;)